File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* current = head;
        int length= 0;
        while(current!=nullptr){
            current = current->next;
            length++;
        }
        int terminate = length - n;

        if (terminate == 0) {
        ListNode* newHead = head->next;
        delete head;
        return newHead;
    }

        current = head;
        int curr_len = 0;
        while(curr_len != terminate - 1){
            curr_len++;
            current = current->next;
        }
        
        ListNode* toDelete = current->next;
    current->next = current->next->next;
    delete toDelete;

    return head;

    }
};

File: Data Structures & Algorithms/add-two-numbers/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        curr1 = l1
        curr2 = l2
        carry = 0
        answer = ListNode()
        currPointer = answer

        while curr1 or curr2 or carry:
            calc = (curr1.val if curr1 else 0) + ( curr2.val if curr2 else 0) + carry
            carry = calc//10
            remain = calc % 10
            currNode = ListNode(val=remain)
            answer.next = currNode
            answer = answer.next
            curr1 = curr1.next if curr1 else 0
            curr2 = curr2.next if curr2 else 0
        
        return currPointer.next

